TradeEngine/src/main/java/com/gala/sam/tradeEngine/service/MarketService.java

```diff
@@ -48,63 +61,115 @@ void init() {
     updateMarketStateFromOrderRepository(marketState, orderRepository);
   }
 
-  public Order enterOrder(OrderReq orderReq) {
-    log.info("Processing Triggered Stop Orders");
+  public Optional<AbstractOrder> enterOrder(AbstractOrderRequest orderRequest) {
+    log.info("Processing Order Time-step with order request: {}", orderRequest);
 
-    Order order = concreteOrderGenerator.getConcreteOrder(orderReq);
+    final AbstractOrder order;
+    try {
+      IOrderValidator<AbstractOrderRequest> orderValidator = orderValidatorFactory.getOrderValidator(orderRequest.getType());
+      List<String> errors = orderValidator.findErrors(orderRequest);
+      if (!errors.isEmpty()) {
+        log.error("Order request could not be validated. Reasons: {}", errors);
+        return Optional.empty();
+      } else {
+        log.debug("Order request was validated: {}", orderRequest);
+      }
+
+      IEnteredOrderGenerator enteredOrderGenerator = concreteOrderGeneratorFactory
+          .getEnteredOrderGenerator(orderRequest.getType());
+
+      order = enteredOrderGenerator.generateConcreteOrder(orderRequest);
+      log.debug("Concrete Order generated with id: {}", order.getOrderId());
+
+    } catch (AbstractOrderFieldNotSupportedException e) {
+      log.error(
+          "Concrete order could not be generated since when handling order request {} an exception was raised: {}",
+          orderRequest, e.getStackTrace());
+      return Optional.empty();
+    }
+
+    try {
+
+      processOrder(order);
 
-    processOrder(order);
-    processTriggeredStopOrders();
-    return order;
+    } catch (AbstractOrderFieldNotSupportedException e) {
+      log.error("The Order {} could not be processed since when handling order an exception was raised: {}", order.getOrderId(), e.getStackTrace());
+      return Optional.empty();
+    }
+
+    try {
+      processTriggeredStopOrders();
+    } catch (ProcessingStopOrderException e) {
+      log.error("Not all stop orders have been handled correctly since an exception was raised in processing them: {}", e.getStackTrace());
+    }
+
+    return Optional.of(order);
   }
 
   public List<Trade> getAllMatchedTrades() {
     return marketState.getTrades();
   }
 
-  private void processOrder(Order order) {
+  private void processOrder(AbstractOrder order)
```

Comment:
Double-check that all methods that call this handle all exceptions that this method might throw.